### PR TITLE
Feature/bphh 384/setup geocoder

### PR DESCRIPTION
**Pull request by bkmaibach**
## Description

Add geocoder and the start of permit application creation.
Includes custom Site Selectro to obtain PID and radio select for the fetched permit and work type options.
Mobile responsive and accessibility maxxed

ALSO remove the Errors constants that were too tedious and not enough benefit. (isolated commit)

## What type of PR is this? (check all applicable)

- [ x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📦 Chore (Release)
- [ ] ✅ Test

## Related Tickets & Documents
https://hous-bssb.atlassian.net/browse/BPHH-384
https://hous-bssb.atlassian.net/browse/BPHH-51
https://hous-bssb.atlassian.net/browse/BPHH-334

## Steps to QA

SEED THE NEW PermitClassification data, the seeder has been updated to update existing records.
``` rails db:seed ```
Log in as submitter

## UI Accessibility Checklist

All Wave passing, had to modify mocks very slightly to get around struggles.

_Note the following checklist is not an exhaustive list. Check
the [WebAIM checklist](https://webaim.org/standards/wcag/checklist) for
a user-friendly checklist with more details. Aim to meet at least *AA* conformance level where applicable. Check
the [WCAG guidelines](https://webaim.org/standards/wcag/checklist) for the full guidelines._

- [ x] Markup uses semantic HTML?
- [x ] Additional context added through `aria-roles` and `aria-labels`?
- [ x] Checked
      with [Wave Browser Extension](https://chromewebstore.google.com/detail/wave-evaluation-tool/jbbplnpkjmmeebjpijfedlgcdilocofh)
      for accessibility errors?
- [x ] Usable with a keyboard and navigable in a logical order?
- [ x] Usable using a screen reader (e.g. Voiceover for Mac) with enough context for the user?

## General Checklist

- [ ] ✅ Provide tests for your changes where relevant
- [ x] 📝 Use descriptive commit messages
- [ ] 📗 Update any related documentation and include any relevant screenshots
- [x ] 📱 For UI-related tasks, ensure responsive design is implemented across multiple screen size
- [ ] 🗂️ Move your relevant story/task to the _Ready for Code Review_ state

## [optional] Are there any post-deployment tasks we need to perform?

SEED THE NEW PermitClassification data, the seeder has been updated to update existing records.
``` rails db:seed ```

### CODE REVIEW

File: app/frontend/stores/permit-classification-store.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      if (self.activityMap.size === 0) yield self.fetchPermitClassifications()
+      return self.activities.map((a) => ({ label: a.name, value: a.id })) as IOption[]
+    }),
+    fetchPermitTypeOptions: flow(function* () {
+      if (self.permitTypeMap.size === 0) yield self.fetchPermitClassifications()
+      return self.permitTypes.map((pt) => ({ label: pt.name, value: pt })) as IOption<IPermitType>[]
+    }),
+    fetchActivityOptions: flow(function* () {
+      if (self.activityMap.size === 0) yield self.fetchPermitClassifications()
+      return self.activities.map((a) => ({ label: a.name, value: a })) as IOption<IActivity>[]
```
Comment (taw27) [inline app/frontend/stores/permit-classification-store.ts:81]: do only want to do this by size. What happens if another user adds a new activity etc while the current user is browsing. They won't get the latest data

File: app/frontend/stores/root-store.ts [hunk 2]
```diff
@@ -15,9 +17,11 @@ export const RootStoreModel = types
     sessionStore: types.optional(SessionStoreModel, {}),
     userStore: types.optional(UserStoreModel, {}),
     permitApplicationStore: types.optional(PermitApplicationStoreModel, {}),
+    permitClassificationStore: types.optional(PermitClassificationStoreModel, {}),
     jurisdictionStore: types.optional(JurisdictionStoreModel, {}),
-    requirementBlockStore: types.optional(RequirementBlockStore, {}),
-    requirementTemplateStore: types.optional(RequirementTemplateStore, {}),
+    RequirementBlockStoreModel: types.optional(RequirementBlockStoreModel, {}),
+    RequirementTemplateStoreModel: types.optional(RequirementTemplateStoreModel, {}),
```
Comment (taw27) [inline app/frontend/stores/root-store.ts:23]: why are these two pascal case? They are instances so should be camel case

**Review by taw27 — CHANGES_REQUESTED** [review]

File: app/controllers/api/geocoder_controller.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
+class Api::GeocoderController < Api::ApplicationController
+  skip_after_action :verify_policy_scoped
+  after_action :verify_authorized
+
+  def site_options
+    authorize :geocoder, :site_options?
+    begin
+      wrapper = Wrappers::Geocoder.new
```
[the remark this answers is not in the document]
Comment (bkmaibach) [reply app/controllers/api/geocoder_controller.rb:8]: What would that entail?

File: app/frontend/stores/permit-classification-store.ts (hunk 1 above)
Comment (bkmaibach) [reply app/frontend/stores/permit-classification-store.ts:81]: we are just checking if the data is loaded yet. this data is not expected to change normaly

File: app/frontend/stores/root-store.ts (hunk 2 above)
Comment (bkmaibach) [reply app/frontend/stores/root-store.ts]: fixed

**Review by taw27 — APPROVED** [review]

**Pull request merged**